crates/ide-assists/src/handlers/add_missing_impl_members.rs

```diff
@@ -181,6 +186,25 @@ fn try_gen_trait_body(
     gen_trait_fn_body(func, &trait_path, &adt.value)
 }
 
+fn in_nested_scope(ctx: &AssistContext<'_>) -> bool {
+    if ctx
+        .token_at_offset()
+        .find(|token| match token.kind() {
+            syntax::SyntaxKind::L_PAREN
+            | syntax::SyntaxKind::R_PAREN
+            | syntax::SyntaxKind::L_CURLY
+            | syntax::SyntaxKind::R_CURLY => true,
+            _ => false,
+        })
+        .is_some()
+    {
+        return false;
```

Comment:
I don't think this is quite correct, since it would incorrectly return `false` and display the assist when in something like

```rust
fn f() {
    async {
        // ...
    }$0
    // ...
}
```